lib/db_connection/ownership/manager.ex

```diff
@@ -98,6 +104,31 @@ defmodule DBConnection.Ownership.Manager do
   end
 
   @impl true
+  def handle_call(:get_connection_metrics, _from, %{pool: pool, owners: owners, log: log} = state) do
+    {:ok, pool_metrics} = DBConnection.ConnectionPool.get_connection_metrics(pool)
+
+    proxy_metrics =
+      owners
+      |> Enum.map(fn {_, {proxy, _, _}} ->
+        if Process.alive?(proxy) do
+          GenServer.call(proxy, :get_connection_metrics)
+        end
+      end)
+      |> Enum.reject(&is_nil/1)
+
+    {:reply, {:ok, pool_metrics ++ proxy_metrics}, state}
+  catch
+    :exit, reason ->
+      if log do
+        Logger.log(
+          log,
+          "Caught :exit while calling :get_connection_metrics due to #{inspect(reason)}"
+        )
+      end
+
+      {:reply, :error, state}
```

Comment:
```suggestion
    {:ok, pool_metrics} = DBConnection.ConnectionPool.get_connection_metrics(pool)

    proxy_metrics =
      owners
      |> Enum.map(fn {_, {proxy, _, _}} ->
        try do
          GenServer.call(proxy, :get_connection_metrics)
        catch
          :exit, _ when log -> 
            if log do
              Logger.log(
                log,
                "Caught :exit while calling :get_connection_metrics due to #{inspect(reason)}"
              )
            end

            nil
        end
      end)
      |> Enum.reject(&is_nil/1)

    {:reply, {:ok, pool_metrics ++ proxy_metrics}, state}
```

Reply:
Here is the suggestion. The poll is linked to the current process, so we don't have to worry about it crashing, cause it would crash us too. There is a chance of timeout, but we can also just pass `:infinity` as the timeout (which I recommend anyway).

The only pending question is: do you want to return `{:ok, metrics}` or just the metrics directly?

Reply:
We can't return just metrics because it's the part of DBConnection.Pool behavior.
By the way, do we really want the client of get_connection_metrics to wait indefinitely for a response or handle timeouts on its own?

Reply:
Actually, we've gotten rid of potential errors. We probably get rid of maybeish type in the public API as well

Reply:
> By the way, do we really want the client of get_connection_metrics to wait indefinitely for a response

Why would we wait indefinitely? A timeout is a safe default if we have blocking code but that's not the case here. The whole pool is design to queue things immediately instead of blocking.

Reply:
I meant the `GenServer.call` calls with `timeout = :infinity` which in theory could cause indefinite wait time